bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -251,7 +255,7 @@ public Rule add(Rule rule) {
      * @param rule candidate for unique ID
      * @return a rule with UID
      */
-    protected Rule initRuleId(String rUID, Rule rule) {
+    protected @Nonnull Rule initRuleId(String rUID, Rule rule) {
```

Comment:
>>> but it adds additional checks at runtime

>> I didn't get the point why the annotation itself will add checks at runtime.
Could you please explain.

> Because the @nonnull is annotated with @retention(RetentionPolicy.RUNTIME)

Can you please explain me about the relationship between the retention policy runtime and additional checks at runtime?

```
Annotations are to be recorded in the class file by the compiler and retained by the VM at run time, so they may be read reflectively.
```

So, a point could be "the file size of the classes increases by a few bytes".